controllers/users/authenticate-web.js

```diff
@@ -22,44 +22,67 @@
    * @type {RegisterBody}
    */
   const users = req.body;
-  users.username = 'auth_web_' + moment().format('YYYYMMDDHHmmssSSS');
+
+  const checkUser = await User.findOne({
+    where: {
+      human_id: users.human_id
+    }
+  });
 
   let insertedObject = {};
 
-  /**
-   * Inserting data to Postgres DB
-   */
-  try {
-    insertedObject = await sequelize.transaction(async (t) => {
-      const user = await UsersFunction.register(User, users, t);
-      const anonymousUser = await UsersFunction.registerAnonymous(User, user?.user_id, t);
+  if (checkUser) {
+    insertedObject = {
+      user: checkUser,
+      anonymousUser: await UsersFunction.findAnonymousUserId(User, checkUser.dataValues.user_id, {
+        raw: false
+      })
+    };
+  } else {
+    users.username = 'auth_web_' + moment().format('YYYYMMDDHHmmssSSS');
+    users.verified_status = 'UNVERIFIED';
+    /**
+     * Inserting data to Postgres DB
+     */
+    try {
+      insertedObject = await sequelize.transaction(async (t) => {
+        const user = await UsersFunction.register(User, users, t);
+        const anonymousUser = await UsersFunction.registerAnonymous(User, user?.user_id, t);
 
-      return {
-        user,
-        anonymousUser
-      };
-    });
-  } catch (e) {
-    console.log('error on sql transaction', e);
-    return res.status(500).json({
-      status: 'error on sql transaction',
-      code: 500,
-      message: e
-    });
+        return {
+          user,
+          anonymousUser
+        };
+      });
+    } catch (e) {
+      console.log('error on sql transaction', e);
+      return res.status(500).json({
+        status: 'error on sql transaction',
+        code: 500,
+        message: e
+      });
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling here is good but logging the error to the console isn't a best practice for production code. It would be better to use a dedicated logging service that can provide more context and control over what gets logged and where those logs get sent. Also, sending the raw error message back to the client might expose sensitive information or implementation details. Consider returning a generic error message to the client and logging the detailed error server-side.

```diff
-      console.log('error on sql transaction', e);
-      return res.status(500).json({
-        status: 'error on sql transaction',
-        code: 500,
-        message: e
-      });
+      logger.error('error on sql transaction', e);
+      return res.status(500).json({
+        status: 'error on sql transaction',
+        code: 500,
+        message: 'An error occurred while processing your request.'
+      });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->